website/preprints/tasks.py

```diff
@@ -0,0 +1,153 @@
+import datetime
+import logging
+import urlparse
+
+import requests
+
+from framework.celery_tasks import app as celery_app
+
+from website import settings
+
+logger = logging.getLogger(__name__)
+
+@celery_app.task(ignore_results=True)
+def on_preprint_updated(preprint_id):
+    # WARNING: Only perform Read-Only operations in an asynchronous task, until Repeatable Read/Serializable
+    # transactions are implemented in View and Task application layers.
+    from website.models import PreprintService
+    preprint = PreprintService.load(preprint_id)
+
+    if settings.SHARE_URL and settings.SHARE_API_TOKEN:
+        resp = requests.post('{}api/v2/normalizeddata/'.format(settings.SHARE_URL), json={
+            'created_at': datetime.datetime.utcnow().isoformat(),
+            'normalized_data': {
+                '@graph': format_preprint(preprint)
+            },
+        }, headers={'Authorization': 'Bearer {}'.format(settings.SHARE_API_TOKEN)})
+        logger.debug(resp.content)
+        resp.raise_for_status()
+
+def format_institution(institution):
+    return [{
+        '@id': '_:{}'.format(institution._id),
+        '@type': 'institution',
+        'name': institution.title,
+    }]
+
+
+def format_user(user):
+    return sum([[{
+        '@id': '_:{}'.format(user._id),
+        '@type': 'person',
+        'suffix': user.suffix,
+        'given_name': user.given_name,
+        'family_name': user.family_name,
+        'additional_name': user.middle_names,
+    }, {
+        '@id': '_:throughidentifier-{}'.format(user._id),
+        '@type': 'throughidentifiers',
+        'person': {
+            '@id': '_:{}'.format(user._id),
+            '@type': 'person',
+        },
+        'identifier': {
+            '@id': '_:identifier-{}'.format(user._id),
+            '@type': 'identifier',
+        }
+    }, {
+        '@id': '_:identifier-{}'.format(user._id),
+        '@type': 'identifier',
+        'url': urlparse.urljoin(settings.DOMAIN, user.profile_url),
+        'base_url': settings.DOMAIN
+    }]] + [[{
+        '@id': '_:{}-{}'.format(user._id, institution._id),
+        '@type': 'affiliation',
+        'entity': {
+            '@id': '_:{}'.format(institution._id),
+            '@type': 'institution',
+        },
+        'person': {
+            '@id': '_:{}'.format(user._id),
+            '@type': 'person',
+        }
+    }] + format_institution(institution) for institution in user.affiliated_institutions], [])
+
+def format_contributor(preprint, user):
+    return [{
+        '@id': '_:{}-{}'.format(preprint._id, user._id),
+        '@type': 'contributor',
+        'person': {'@id': '_:{}'.format(user._id), '@type': 'person'},
```

Comment:
Added in https://github.com/CenterForOpenScience/osf.io/pull/6346/commits/268a7a9a7db6b7ea5a5c23d41c154507519f714f